src/toast/util/toastReducer.ts

```diff
@@ -57,6 +64,29 @@ function toastReducer(state: ToastState, action: ToastAction): ToastState {
       break;
     }
 
+    case "SET_LIMIT": {
+      const {limit} = action;
+      // prune the toasts exceeding the limit
+      const toastStack = state.toastStack.slice(
+        Math.max(0, state.toastStack.length - limit)
+      );
```

Comment:
```suggestion
      if (limit && toastStack.length > limit) {
        // prune the toasts exceeding the limit
        toastStack = toastStack.slice(toastStack.length - limit);
      }
```

We can use the approach you used in "DISPLAY", I think that is more expressive